Parsing Datalog fails when parenthesis are used in expressions

The following code throws a panic for null pointer dereference [1]: 
```
rule, _ := parser.FromStringRule(`var($a) <- user($a), !($a == "abc")`)
```

I did a bit of digging around and it appears that `ruleParser.ParseString`[2] does not properly parse the expression in parenthesis, meaning that the call to `ToBiscuit` [here](https://github.com/biscuit-auth/biscuit-go/blob/551310c71ce0dc904fb0cc67196dbd1cadca5cc6/parser/parser.go#L109) fails. 




[1]
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x603306]

goroutine 1 [running]:
github.com/biscuit-auth/biscuit-go/parser.(*ExprTerm).ToExpr(0x4, 0xc00041fb38)
        /home/ubuntu/go/pkg/mod/github.com/biscuit-auth/biscuit-go@v1.0.0/parser/grammar.go:278 +0x46
github.com/biscuit-auth/biscuit-go/parser.(*Expr5).ToExpr(0xc000400900, 0x0)
        /home/ubuntu/go/pkg/mod/github.com/biscuit-auth/biscuit-go@v1.0.0/parser/grammar.go:261 +0x29
github.com/biscuit-auth/biscuit-go/parser.(*ExprTerm).ToExpr(0x37, 0xc00041fb38)
        /home/ubuntu/go/pkg/mod/github.com/biscuit-auth/biscuit-go@v1.0.0/parser/grammar.go:274 +0xf2
github.com/biscuit-auth/biscuit-go/parser.(*Expr5).ToExpr(0xc000400960, 0x203000)
        /home/ubuntu/go/pkg/mod/github.com/biscuit-auth/biscuit-go@v1.0.0/parser/grammar.go:261 +0x29
github.com/biscuit-auth/biscuit-go/parser.(*Expr4).ToExpr(0xc000400980, 0xc00041fa10)
        /home/ubuntu/go/pkg/mod/github.com/biscuit-auth/biscuit-go@v1.0.0/parser/grammar.go:253 +0x29
github.com/biscuit-auth/biscuit-go/parser.(*Expr3).ToExpr(0xc0004009a0, 0x20)
        /home/ubuntu/go/pkg/mod/github.com/biscuit-auth/biscuit-go@v1.0.0/parser/grammar.go:245 +0x29
github.com/biscuit-auth/biscuit-go/parser.(*Expr2).ToExpr(0xc0004009c0, 0xc00041fa78)
        /home/ubuntu/go/pkg/mod/github.com/biscuit-auth/biscuit-go@v1.0.0/parser/grammar.go:237 +0x29
github.com/biscuit-auth/biscuit-go/parser.(*Expr1).ToExpr(0xc0004009e0, 0x1)
        /home/ubuntu/go/pkg/mod/github.com/biscuit-auth/biscuit-go@v1.0.0/parser/grammar.go:229 +0x29
github.com/biscuit-auth/biscuit-go/parser.(*Expression).ToExpr(0xc000400a00, 0x8951a0)
        /home/ubuntu/go/pkg/mod/github.com/biscuit-auth/biscuit-go@v1.0.0/parser/grammar.go:221 +0x29
github.com/biscuit-auth/biscuit-go/parser.(*Rule).ToBiscuit(0xc0003ece80)
        /home/ubuntu/go/pkg/mod/github.com/biscuit-auth/biscuit-go@v1.0.0/parser/grammar.go:478 +0xf0
github.com/biscuit-auth/biscuit-go/parser.(*parser).Rule(0xc000400440, {0x6804ac, 0x23})
        /home/ubuntu/go/pkg/mod/github.com/biscuit-auth/biscuit-go@v1.0.0/parser/parser.go:109 +0xc5
github.com/biscuit-auth/biscuit-go/parser.FromStringRule({0x6804ac, 0x23})
        /home/ubuntu/go/pkg/mod/github.com/biscuit-auth/biscuit-go@v1.0.0/parser/parser.go:226 +0xb9
```

[2]
https://github.com/biscuit-auth/biscuit-go/blob/551310c71ce0dc904fb0cc67196dbd1cadca5cc6/parser/parser.go#L105

--- a/parser/parser_test.go
+++ b/parser/parser_test.go
@@ -606,3 +606,9 @@ func TestMustParserFact(t *testing.T) {
 // 		})
 // 	}
 // }
+
+func TestIssue84(t *testing.T) {
+	rule, err := FromStringRule(`var($a) <- user($a), !($a == "abc")`)
+	_ = rule
+	require.NoError(t, err)
+}

